api/commands/run.ts

```diff
@@ -247,10 +248,15 @@ export class Runner {
     parentAction.tasks.push(taskResult);
     await this.triggerChange();
     try {
-      const rows = await this.adapter.execute(task.statement, {
-        onCancel: handleCancel => this.eEmitter.on(CANCEL_EVENT, handleCancel),
-        maxResults: 1
-      });
+      // Retry this function a given number of times, configurable by user
+      const rows = await retryPromise(
+        () =>
+          this.adapter.execute(task.statement, {
+            onCancel: handleCancel => this.eEmitter.on(CANCEL_EVENT, handleCancel),
+            maxResults: 1
+          }),
+        this.graph.projectConfig.idempotentActionRetries || 0
```

Comment:
we should *always* have zero retries for any non-idempotent action, i.e. any task of type "operations"

(longer term we will want to let the user mark their actions as idempotent or not, but for now we can assume that operations aren't, everything else is.)

Reply:
Changed line to this:

```
task.type === "operation" ? 0 : this.graph.projectConfig.idempotentActionRetries || 3
```

So operations will never retry.